commands/intent.go

```diff
@@ -0,0 +1,29 @@
+package commands
+
+const (
+	// DefaultIntent Default Intent Type
+	DefaultIntent = iota
+	// VersionIntent Intent to print version
+	VersionIntent = iota
+)
+
+// Intent Used for deriving user intent based on predefined business rules
+type Intent struct {
+	PrintVersion bool
+}
+
+// NewIntent Intent Constructor
+func NewIntent(printVersion bool) Intent {
+	return Intent{
+		PrintVersion: printVersion,
+	}
+}
+
+// Is Get derived intent type
+func (i *Intent) Is() int {
```

Comment:
What does `Is` mean?

Reply:
I think this may be less clear here than when viewed in context (I'm happy to update this if you have better ideas though).

I went with `Is` because I expected the usage to be like:

```
intent.Is() == DefaultIntent // true/false
```

Reply:
I'll probably feel more comfortable making a suggestion after you've given a type to the enum. Maybe that in the "intent" and the struct is like `IntentInfo` or something. In that case, I'd recommend `switch (info.Intent())`

Reply:
I don't **love** `IntentInfo`, but it was the best I couldn't come up with anything better. I *do* think `.Intent()` is more clear.